lib/src/main/java/graphql/nadel/util/MapUtil.kt

```diff
@@ -0,0 +1,38 @@
+package graphql.nadel.util
+
+internal object MapUtil {
+    fun mergeListAtPath(
+        originalMap: Map<String, Any>,
+        partitionedMaps: List<Map<String, Any?>>,
+        path: List<String>,
+    ): Map<String, Any>? {
+
+        val immutableMap = originalMap.toMutableMapRecursively()
+        var tempOriginal: MutableMap<String, Any> = immutableMap
+        var tempPartitioned: List<Map<*, *>> = partitionedMaps
+
+        path.forEachIndexed { idx, value ->
+            if (tempOriginal[value] is Map<*, *>) {
+                tempOriginal = tempOriginal[value] as MutableMap<String, Any>
+                tempPartitioned = tempPartitioned.map { it[value] as Map<*, *> }
+            } else if (path.size - 1 == idx && tempOriginal[value] is List<*>) {
+                tempOriginal[value] =
+                    (tempOriginal[value] as List<*>) + tempPartitioned.flatMap { it[value] as List<*> }
+            } else {
+                return null
+            }
+        }
+        return immutableMap
+    }
+
+    fun <K, V> Map<K, V>.toMutableMapRecursively(): MutableMap<K, V> {
```

Comment:
If it's from Jackson it should always be mutable.

So you just force cast it to the mutable version of the type.